satellite/console/service.go

```diff
@@ -155,6 +166,21 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 			return errs.New(internalErrMsg)
 		}
 
+		if currentReward != nil {
+			// User can only earn credits after activating their account. Therefore, we set the credits to 0 on registration
+			newCredit := UserCredit{
+				UserID:        u.ID,
+				OfferID:       currentReward.ID,
+				ReferredBy:    nil,
+				CreditsEarned: currency.Cents(0),
+				ExpiresAt:     time.Now().UTC().AddDate(0, 0, currentReward.InviteeCreditDurationDays),
+			}
+
+			err = tx.UserCredits().Create(ctx, newCredit)
+			if err != nil {
+				return err
```

Comment:
should this error be wrapped?

Reply:
Looks like the `Create` method there always returns errs-created error instances anyway, so no need to wrap them for the purpose of attaching a traceback. But I'm not entirely sure what our policy is. Do we always wrap anyway, unless there are error values that break if wrapped? It's getting confusing to me

Reply:
I'm not sure what our policy exactly is for wrapping errors either. I found this styling guide in our doc https://github.com/storj/docs/blob/master/code/Style.md#error-handling

Reply:
the real policy is that if an error isn't already always a zeebo/errs generated error, it should be wrapped. so the answer is contextual. the main point of zeebo/errs for us is to add stack traces, so if there isn't a stack trace attached to the error yet, wrapping adds one. there's one exception to this as egon recently pointed out which is errors that have very specific behavior associated such as io.EOF. but that won't happen here, and i believe we've extended dbx to use zeebo/errs, so i don't think you need to wrap here. on the flip side, because this won't be io.EOF, it also won't hurt to wrap here.

Reply:
Additionally, if you don’t care about adding a class to the error, using `errs.Wrap` only adds a stack trace if one does not already exist. (Class wrapping also has this property, and `errs.Wrap` can be thought of wrapping with no class.) The intent of that function is to make the decision easy for readers and writers: always call `errs.Wrap` unless you care about io.EOF as described above.

Reply:
I will add and don't wrap if the error is a GRPC one and it has to be sent through the wire